client/apps_test.go

```diff
@@ -283,60 +343,85 @@ func (cs *clientSuite) TestClientServiceStop(c *check.C) {
 	cs.status = 202
 	cs.rsp = `{"type": "async", "status-code": 202, "change": "24"}`
 
-	type tT struct {
-		names   []string
-		opts    client.StopOptions
-		comment check.CommentInterface
-	}
-
-	var scs []tT
-
-	for _, names := range [][]string{
-		nil, {},
-		{"foo"},
-		{"foo", "bar", "baz"},
-	} {
-		for _, opts := range []client.StopOptions{
-			{Disable: true},
-			{Disable: false},
-		} {
-			scs = append(scs, tT{
-				names:   names,
-				opts:    opts,
-				comment: check.Commentf("{%q; %#v}", names, opts),
-			})
-		}
+	tests := []struct {
+		names []string
+		scope client.ScopeSelector
+		users client.UserSelector
+		opts  client.StopOptions
+	}{
+		{},
+		{
+			opts: client.StopOptions{
+				Disable: true,
+			},
+		},
+		{
+			names: []string{"foo"},
+		},
+		{
+			names: []string{"foo"},
+			opts: client.StopOptions{
+				Disable: true,
+			},
+		},
+		{
+			names: []string{"foo", "bar", "baz"},
+		},
+		{
+			names: []string{"foo", "bar", "baz"},
+			opts: client.StopOptions{
+				Disable: true,
+			},
+		},
+		{
+			names: []string{"foo"},
+			scope: []string{"user"},
+		},
+		{
+			names: []string{"foo"},
+			scope: []string{"system"},
+		},
+		{
+			names: []string{"foo"},
+			scope: []string{"system", "user"},
+		},
+		{
+			names: []string{"foo"},
+			users: client.UserSelector{
+				Selector: client.UserSelectionSelf,
+			},
+		},
+		{
+			names: []string{"foo"},
+			users: client.UserSelector{
+				Selector: client.UserSelectionAll,
+			},
+		},
 	}
 
-	for _, sc := range scs {
-		id, err := cs.cli.Stop(sc.names, sc.opts)
+	for _, sc := range tests {
+		comment := check.Commentf("{%q; %q; %q; %#v}", sc.names, sc.scope, sc.users, sc.opts)
```

Comment:
btw. you can also `c.Logf("testing ... ")`, IIRC the logs are displayed if a test fails